posthog/src/main/java/com/posthog/internal/PostHogRemoteConfig.kt

```diff
@@ -638,6 +654,103 @@ public class PostHogRemoteConfig(
         }
     }
 
+    public fun setPersonPropertiesForFlags(properties: Map<String, Any>) {
+        synchronized(personPropertiesForFlagsLock) {
+            personPropertiesForFlags.putAll(properties)
+            config.cachePreferences?.setValue(
+                PostHogPreferences.PERSON_PROPERTIES_FOR_FLAGS,
+                personPropertiesForFlags as Any,
+            )
+        }
+    }
+
+    public fun resetPersonPropertiesForFlags() {
+        synchronized(personPropertiesForFlagsLock) {
+            personPropertiesForFlags.clear()
+            config.cachePreferences?.remove(PostHogPreferences.PERSON_PROPERTIES_FOR_FLAGS)
+        }
+    }
+
+    public fun setGroupPropertiesForFlags(
+        groupType: String,
+        properties: Map<String, Any>,
+    ) {
+        synchronized(groupPropertiesForFlagsLock) {
+            val existing = groupPropertiesForFlags.getOrPut(groupType) { mutableMapOf() }
+            existing.putAll(properties)
+            config.cachePreferences?.setValue(
+                PostHogPreferences.GROUP_PROPERTIES_FOR_FLAGS,
+                groupPropertiesForFlags as Any,
+            )
+        }
+    }
+
+    public fun resetGroupPropertiesForFlags(groupType: String? = null) {
+        synchronized(groupPropertiesForFlagsLock) {
+            if (groupType != null) {
+                groupPropertiesForFlags.remove(groupType)
+                config.cachePreferences?.setValue(
+                    PostHogPreferences.GROUP_PROPERTIES_FOR_FLAGS,
+                    groupPropertiesForFlags as Any,
+                )
+            } else {
+                groupPropertiesForFlags.clear()
+                config.cachePreferences?.remove(PostHogPreferences.GROUP_PROPERTIES_FOR_FLAGS)
+            }
+        }
+    }
+
+    private fun getPersonPropertiesForFlags(): Map<String, Any> {
```

Comment:
i think this whole class should use `<String, Any?>` here as well for user props and group props

Reply:
This should be fine considering we're only caching properties for usage with the flags API. `null` values _should_ be stripped out by serialization anyway.